docs/articles/documentation/guides/concepts/typescript-and-coffeescript.md

```diff
@@ -42,6 +42,60 @@ Whenever TestCafe encounters TypeScript compilation errors, it includes correspo
 > and [selector.addCustomMethods](../../reference/test-api/selector/addcustommethods.md)
 > sections to learn how to extend selectors in TypeScript.
 
+#### Type Cast Page Elements
+
+When you identify HTML elements in your TypeScript [client-side code](../basic-guides/obtain-client-side-info.md), you need to type cast the elements. To use properties of the basic [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement), cast the element to the `HTMLElement` type.
+
+An example below uses [Element.scrollIntoView()](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) inside a [client function](../../reference/test-api/clientfunction/README.md) to scroll an HTML element into view.
+
+```ts
+import { ClientFunction, Selector } from 'testcafe';
+
+const scrollIntoView = ClientFunction( (selector: Selector) => {
+    const element = selector() as unknown as HTMLElement;
+    element.scrollIntoView();
+});
+
+fixture`HTMLElement`
+    .page('https://example.com');
+
+test('Scroll element into view', async t => {
+    const bottomOfPage = Selector('#bottom-div');
+
+    await scrollIntoView(bottomOfPage);
+});
+```
+
+To use members of a particular HTML element interface (for example, [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement)), cast the element to that specific type.
+
+The example below uses [t.eval](../../reference/test-api/testcontroller/eval.md) to determine if an ordered list has the [*reversed* property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties). Because the property exists on the [HTMLOListElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties), but not on the [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#properties), a narrower type casting is needed.
```

Comment:
Very confusing. Also, only the type itself can be narrow - not the casting.

```suggestion
The example below calls the [t.eval](../../reference/test-api/testcontroller/eval.md) method to determine if an ordered list is [reversed](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties). The generic [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#properties) cannot access the DOM element's `reversed` property, so the `list` object is assigned the [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties) data type.
```

Reply:
> The generic HTMLElement Interface cannot access the DOM element's `reversed` property

An interface can't access a property. A user might access a property defined on an interface.

```suggestion
The code in an example below calls the [t.eval](../../reference/test-api/testcontroller/eval.md) method to determine if an ordered list is [reversed](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties). The generic [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#properties) does not have a `reversed` property, so it is necessary to cast the [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties) data type to the `list` object to avoid a compilation error.
```

Reply:
Almost! Just a couple notes. 

>An interface can't access a property.
agreed, although it can provide access to a property.
> A user might access a property defined on an interface.
not necessarily - the property belongs to an object, the interface is merely a mediator. Lists have the `reversed` property whether or not we can access it.

>cast the type to the list
It's actually the opposite. You cast the list to be of a certain type. see [this](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions) example. If possible, I'd completely avoid using the word "cast" since it's *not* the official term for this process. The TypeScript handbook uses the verb "to convert" interchangeably 

```suggestion
The example code below calls the [t.eval](../../reference/test-api/testcontroller/eval.md) method to determine if an ordered list is [reversed](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties). The generic [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#properties) does not provide access to the element's `reversed` property. Therefore it is necessary to convert the `list` object to the [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties) data type.
```

Reply:
> It's actually the opposite

You are right, that's a typo

Just one thing: 

```suggestion
The example code below calls the [t.eval](../../reference/test-api/testcontroller/eval.md) method to determine if an ordered list is [reversed](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties). The generic [HTMLElement Interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#properties) does not provide access to the element's `reversed` property. To avoid compilation errors, it is necessary to convert the `list` object to the [HTMLOListElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOListElement#properties) data type.
```